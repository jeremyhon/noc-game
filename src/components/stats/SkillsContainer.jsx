import React from 'react'
import PropTypes from 'prop-types'
import _ from 'lodash'
import classnames from 'classnames'

import { skills } from 'lib/data'
import skillsPropType from './skillsPropType'
import './StatsContainer.css'

const renderSkills = (skills) => _.map(skills,
  (skill, key) => (
    <p key={key}>
      <span className="stat-name">{_.startCase(key)}:</span>
      <span className="stat-value">{skill}</span>
    </p>),
)

export const SkillsContainer = (props) => (
  <div className={classnames(props.className, 'stat-container')} >
    <h3 className='stat-subtitle'>{props.subtitle}: </h3>
    {renderSkills(_.pick(props.skills, skills))}
  </div>
)

SkillsContainer.defaultProps = {
  className: '',
}

SkillsContainer.propTypes = {
  className: PropTypes.string,
  skills: skillsPropType.isRequired,
  subtitle: PropTypes.string.isRequired,
}

export default SkillsContainer
