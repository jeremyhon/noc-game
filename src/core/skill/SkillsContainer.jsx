import React, { PropTypes } from 'react'
import map from 'lodash/map'
import skillPropType from '../../lib/skill'

const renderSkills = (skills) => map(skills, (skill, key) => (<li key={key}>{skill}</li>))

const SkillsContainer = (props) => (
  <ul>
    {renderSkills(props.skills)}
  </ul>
)

SkillsContainer.propTypes = {
  skills: PropTypes.arrayOf(skillPropType).isRequired,
}

export default SkillsContainer
