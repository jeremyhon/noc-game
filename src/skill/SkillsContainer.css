.skill-container p {
  margin: 0;
  display: flex; }

.skill-container span {
  text-align: left;
  width: 50%; }

.skill-container .skill-name {
  padding-left: 5em; }

.skill-container .skill-value {
  padding-left: 1em; }
