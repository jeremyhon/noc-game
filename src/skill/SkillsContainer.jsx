import React from 'react'
import map from 'lodash/map'
import skillsPropType from '../skill/multiPropType'

const renderSkills = (skills) => map(skills, (skill, key) => (<li key={key}>{skill}</li>))

const SkillsContainer = (props) => (
  <ul>
    {renderSkills(props.skills)}
  </ul>
)

SkillsContainer.propTypes = {
  skills: skillsPropType.isRequired,
}

export default SkillsContainer
