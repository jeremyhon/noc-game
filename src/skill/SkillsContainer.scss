.skill-container{
  p {
    margin: 0;
    display: flex;
  }

  span {
    text-align: left;
    width: 50%;
  }

  .skill-name {
    padding-left: 5em;
  }

  .skill-value {
    padding-left: 1em;
  }
}
