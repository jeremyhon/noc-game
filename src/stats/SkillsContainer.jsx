import React from 'react'
import PropTypes from 'prop-types'
import map from 'lodash/map'
import startCase from 'lodash/startCase'
import classnames from 'classnames'
import skillsPropType from './skillsPropType'
import './StatsContainer.css'

const renderSkills = (skills) => map(skills,
  (skill, key) => (
    <p key={key}>
      <span className="stat-name">{startCase(key)}:</span>
      <span className="stat-value">{skill}</span>
    </p>),
)

const SkillsContainer = (props) => (
  <div className={classnames(props.className, 'stat-container')} >
    {renderSkills(props.skills)}
  </div>
)

SkillsContainer.defaultProps = {
  className: '',
}

SkillsContainer.propTypes = {
  className: PropTypes.string,
  skills: skillsPropType.isRequired,
}

export default SkillsContainer
