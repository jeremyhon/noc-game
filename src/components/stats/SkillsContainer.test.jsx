import React from 'react'
import { shallow } from 'enzyme'

import SkillsContainer from './SkillsContainer'

it("matches the snapshot", () => {
  const skills = {
    business: 1,
    design: 1,
    technical: 1,
    lulz: 1,
  }
  const wrapper = shallow(<SkillsContainer skills={skills} subtitle="Skills" />)
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).not.toIncludeText('Lulz')
})
