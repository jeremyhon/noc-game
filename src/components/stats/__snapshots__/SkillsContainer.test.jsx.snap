// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="stat-container"
>
  <h3
    className="stat-subtitle"
  >
    Skills
    : 
  </h3>
  <p>
    <span
      className="stat-name"
    >
      Business
      :
    </span>
    <span
      className="stat-value"
    >
      1
    </span>
  </p>
  <p>
    <span
      className="stat-name"
    >
      Design
      :
    </span>
    <span
      className="stat-value"
    >
      1
    </span>
  </p>
  <p>
    <span
      className="stat-name"
    >
      Technical
      :
    </span>
    <span
      className="stat-value"
    >
      1
    </span>
  </p>
</div>
`;
